### PR TITLE
Fix:url.Values is not safe map

### PR DESCRIPTION
**Pull request by hxmhlt**
**What this PR does**:
Url.Values is not safe map, so add the lock to avoid concurrent map read and map write error

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by gaoxinge** [comment]
Does this fix https://github.com/apache/dubbo-go/issues/146?

File: common/url.go [hunk 1]
```diff
@@ -283,14 +286,18 @@ func (c URL) Service() string {
...
 func (c *URL) AddParam(key string, value string) {
+	c.paramsLock.Lock()
 	c.Params.Add(key, value)
+	c.paramsLock.Unlock()
 }
 
 func (c URL) GetParam(s string, d string) string {
 	var r string
+	c.paramsLock.RLock()
 	if r = c.Params.Get(s); r == "" {
```
Comment (AlexStocks) [inline common/url.go]: 'len(r) == 0' instead.

File: common/url.go [hunk 2]
```diff
@@ -64,11 +65,13 @@ func (t RoleType) Role() string {
...
-	Ip           string
-	Port         string
-	Params       url.Values
+	Protocol string
+	Location string // ip+port
+	Ip       string
+	Port     string
+	Params   url.Values
+	//url.Values is not safe map, add to avoid concurrent map read and map write error
+	paramsLock   sync.RWMutex
```
Comment (AlexStocks) [inline common/url.go]: pls place the  paramsLock before Params

**Comment by AlexStocks** [comment]
LGTM

**Pull request merged**